docs/data.rst

```diff
@@ -118,20 +134,82 @@ In an Output, sequences are automatically resized:
 
 .. testcode::
 
-    output.instance.set_number("my_int_sequence[1]", 10)
+    output.instance.set_number("my_int_sequence[5]", 10) # length is now 5
+    output.instance.set_number("my_int_sequence[4]", 9) # length still 5
 
-For a non-primitive sequence:
+You can clear a sequence:
 
 .. testcode::
 
-    output.instance.set_number("my_sequence[1].x", 10)
+    output.instance.clear_member("my_int_sequence") # my_int_sequence is now empty
 
-To get the length of a sequence (Input only):
+To get the length of a sequence in an Input sample:
 
 .. testcode::
 
     length = input[0].get_number("my_sequence#")
 
+Accessing optional members
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Optional members may have a value or not. They are accessed the same way as
```

Comment:
Worth adding a sentence about a use-case? Could be confusing if we only inform people that optional members may or may not have a value

Reply:
I will link to the Users Manual.